### PR TITLE
RN-583: Enable DHIS data group metadata pull

### PR DESCRIPTION
**Pull request by chris-pollard**
## Changes:
- Add metadataPull for dataGroup in DHIS service in Data Broker package.
- Add Data Group fetcher in DHIS Api.

### CODE REVIEW

**Comment by chris-pollard** [comment]
Note this first commit is only for showing other devs, it is not ready for testing.

File: packages/data-broker/src/DataBroker.js [hunk 1]
```diff
@@ -246,9 +246,12 @@ export class DataBroker {
     if (countDistinct(dataSources, 'service_type') > 1) {
       throw new Error('Cannot pull metadata for data sources belonging to different services');
     }
-    const service = this.createService(dataSources[0].service_type);
+    const { config: dataGroupConfig, service_type: serviceType } = dataSources[0];
+
+    const optionsWithConfig = { ...options, dataServices: [dataGroupConfig] };
+    const service = this.createService(serviceType);
```
Comment (rohan-bes) [inline packages/data-broker/src/DataBroker.js:252]: Why is this needed?

File: packages/data-broker/src/services/dhis/DhisService.js [hunk 2]
```diff
@@ -164,12 +170,17 @@ export class DhisService extends Service {
   }
 
   async pullMetadata(dataSources, type, options) {
-    const { organisationUnitCode: entityCode, dataServices = DEFAULT_DATA_SERVICES } = options;
+    const { organisationUnitCodes: entityCodes, dataServices = DEFAULT_DATA_SERVICES } = options;
```
Comment (rohan-bes) [inline packages/data-broker/src/services/dhis/DhisService.js:173]: Do we still use `organisationUnitCode` anywhere? If so should this for backwards compatability:
```suggestion
    const { organisationUnitCode: entityCode, organisationUnitCodes: entityCodes, dataServices = DEFAULT_DATA_SERVICES } = options;
    const dataServiceEntityCode = entityCode || entityCodes[0];
```

File: packages/dhis-api/src/DhisApi.js [hunk 3]
```diff
@@ -547,6 +547,33 @@ export class DhisApi {
...
+  async fetchDataGroup(dataGroupCode, dataElementCodes, includeOptions) {
+    if (!dataGroupCode) {
+      throw new Error('Please provide a data group code');
+    }
+
+    const dataGroups = await this.getRecords({
+      type: PROGRAM,
+      codes: [dataGroupCode],
+      fields: ['code', 'name'],
+    });
```
Comment (rohan-bes) [inline packages/dhis-api/src/DhisApi.js:559]: Would `getRecord` work here? If so it makes things a little clearer 👍

**Review by rohan-bes — COMMENTED** [review]
Looking good at a high level @chris-pollard! Nice work.

Will re-review once you're ready 👍

File: packages/data-broker/src/DataBroker.js (hunk 1 above)
Comment (chris-pollard) [reply packages/data-broker/src/DataBroker.js:252]: Not sure which line you're referring to, but the config is required to get isDataRegional value for creating api.

**Review by rohan-bes — APPROVED** [review]
Nice work @chris-pollard!

Great job implementing this missing spec in the DhisApi.
